### PR TITLE
ref(face-landmarks): convert to typescript and add detector class

### PR DESCRIPTION
**Pull request by gabiborlea**
Fixing [features/face-landmarks] Status error: 401" and "[features/face-landmarks] InvalidStateError: The associated Track is in an invalid state.
And Typescript ref.
Move detector logic from actions to separate class

### CODE REVIEW

File: react/features/face-landmarks/actions.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
-// @flow
-import 'image-capture';
```
Comment (robertpin) [inline react/features/face-landmarks/actions.ts:2]: Is the polyfill not needed anymore?

File: react/features/face-landmarks/middleware.ts [hunk 2]
```diff
@@ -31,7 +34,7 @@ MiddlewareRegistry.register(({ dispatch, getState }) => next => action => {
         // allow using remote face centering data when local face centering is not enabled
         action.conference.on(
             JitsiConferenceEvents.ENDPOINT_MESSAGE_RECEIVED,
-            (participant, eventData) => {
+            (participant: any, eventData: any) => {
```
Comment (robertpin) [inline react/features/face-landmarks/middleware.ts:37]: We have an interface `Participant` so you can use `Participant | undefined`

**Review by robertpin — CHANGES_REQUESTED** [review]

File: react/features/face-landmarks/actions.ts (hunk 1 above)
Comment (gabiborlea) [reply react/features/face-landmarks/actions.ts:2]: Seemed to work without it, but I'll add it back just to be sure

File: react/features/face-landmarks/actionTypes.ts [hunk 3]
```diff
@@ -29,30 +29,23 @@ export const ADD_TO_FACE_EXPRESSIONS_BUFFER = 'ADD_TO_FACE_EXPRESSIONS_BUFFER';
...
-  * {
-  *      type: STOP_FACE_LANDMARKS_DETECTION
-  * }
-  */
-export const STOP_FACE_LANDMARKS_DETECTION = 'STOP_FACE_LANDMARKS_DETECTION';
+export const UPDATE_FACE_COORDINATES = 'UPDATE_FACE_COORDINATES';
 
 /**
- * Redux action type dispatched in order to update coordinates of a detected face.
+ * Redux action type dispatched in order to signal new ace coordinates were obtained for the local participant.
```
Comment (robertpin) [inline react/features/face-landmarks/actionTypes.ts:43]: Typo. `signal new face`

**Review by robertpin — APPROVED** [review]

**Pull request merged**